### ISSUE TITLE
Create a oh-my-fish theme

### ISSUE THREAD

**Issue by bpinto** [fish-shell/oh-my-fish#517]
We should have a default cool theme for omf. Using our logo on the theme would be a big plus.

**Comment by bobthecow** [comment]
```
ln -s ./bobthefish ./omf
```

;)

**Comment by bucaran** [comment]
@bpinto Seriously, why not? We should all pair-review the code, find bugs, improve whatever we can, improve comments.

@bobthecow What do you think about changing the name to just `bob` or `default` or `omf`?

First we need to make sure to download the powerline patched fonts, some like:

```sh
  fonts_dir="$HOME/Library/Fonts"
  powerline=$(mktemp -d $HOME/XXXX)
  git clone "https://github.com/powerline/fonts" $powerline
  find $powerline -name "*.[o,t]tf" | while read font
    do
      cp "$font" "$fonts_dir"
    done
  rm -rf $powerline
```

**Comment by bpinto** [comment]
I know many (everyone? :D) use `bobthefish`, but I don't know if it should be the default.

The problem that I have with it is that it is not blazing fast as it has a lot of information to display, and the reason why I moved to fish was for its speed.

I'd rather have a very simple theme as the default.

**Comment by bpinto** [comment]
Download power line fonts? I can see that as a plugin, but not as part of omf's bootstrap.

**Comment by bucaran** [comment]
> I'd rather have a very simple theme as the default.

I can understand that. Should it be powerline style or plain?

**Comment by bpinto** [comment]
Is there a way to check if a font is patched?

**Comment by bucaran** [comment]
We can check whether fonts are installed.

**Comment by bobthecow** [comment]
But we can't tell whether a shell is currently using it :(

**Comment by bucaran** [comment]
In that case, we need a `fonts` plugin that does nothing, but to download the fonts if they are not already installed.

**Comment by bucaran** [comment]
@bpinto How should the theme look like?

One more thing. @bobthecow Can you go over: https://github.com/fish-shell/oh-my-fish/tree/master/lib/git and see if everything is alright?

**Comment by bpinto** [comment]
@bucaran A plugin to download the font is not bad... many have trouble with this.
Regarding the default theme, I don't know yet. I am bad with design :D

**Comment by tannhuber** [comment]
I think the default theme is most important for new fish users. Experienced people probably use a different theme. So there's no need to implement complex functionality that is not obvious for former bash or zsh users. However, I think it's a good idea to use impressive colors in order to show: »Oh-my-fish can if you want.«

**Comment by bucaran** [comment]
@tannhuber Yes. Should it be powerline or bobbyrussel style?

**Comment by bobthecow** [comment]
For new users, powerline-style is probably not the best because we can almost guarantee that they don't have font support. So they'll end up with undefined unicode squares everywhere, and think it's just ugly :)

**Comment by bucaran** [comment]
>  So they'll end up with undefined unicode squares everywhere, and think it's just ugly :)

Which would be even worse than some having a cool theme.

What about setting the default theme to `russell` instead?

**Comment by bpinto** [comment]
Isn't that oh my zsh-ish?

**Comment by tannhuber** [comment]
@bpinto: Yes, it is.

From my point of view:
- no powerline theme for the reasons already mentioned
- not a simple copy of a theme from another shell, oh-my-fish should have its own superior looking default theme (somebody must decide what this means)
- not a theme that needs vi-mode because it would be frustrating for unexperienced users
- the prompt should be well emphasized in order to distinguish it clearly from shell in- and output

Perhaps the »ocean« theme could build the root of a default theme.

**Comment by bucaran** [comment]
I thought the ocean theme was powerline and had that little anchor unicode symbol?

**Comment by tannhuber** [comment]
@bucaran You're right, the unicode symbols – I think there're two – could be replaced. That's why I wrote it could build the root of a default theme.

**Comment by nesl247** [comment]
I'll actually vote for just using `robbyrussell`. Yes it's from ZSH, but honestly, it's a nice and simple default theme. During the install we can simply add some messages for the user like "Hey, don't forget to check out the variety of cool themes and plugins! Run `omf theme` and `omf install` to try them out!"

**Comment by arvids** [comment]
> From my point of view:
> 
> no powerline theme for the reasons already mentioned
> not a simple copy of a theme from another shell, oh-my-fish should have its own superior looking default theme (somebody must decide what this means)
> not a theme that needs vi-mode because it would be frustrating for unexperienced users
> the prompt should be well emphasized in order to distinguish it clearly from shell in- and output

As a beginner with not that much experience with fish or zsh, this sounds great.
It would also be great if there was friendly way to 'turn of' certain aspects of the theme, for example right_prompt, displaying time or git functionality.

**Comment by bucaran** [comment]
We could change the name and tweak other details, but have a look at:

+ https://github.com/wa/theme-hulk
+ https://github.com/wa/theme-batman
+ https://github.com/wa/theme-flash

**Comment by tannhuber** [comment]
:grinning: And the winner is...

I really like discussions like that.

@bucaran I like these themes as well, especially hulk and batman.

**Comment by bpinto** [comment]
This is what I have in mind: https://github.com/oh-my-fish/theme-default

:eyes:

**Comment by jonscottclark** [comment]
I like it (esp. the little fish), but I think having the full path would be good to have for a default theme.

**Comment by nesl247** [comment]
That looks really nice, though I prefer the colors from `toaster` and `robbyrussell` more.

**Comment by bpinto** [comment]
@jonscottclark I wouldn't mind to be honest, as long as it's configurable and I can disable it. But if most people prefer the full path, than it certainly should be the default.

@nesl247 could you try it on your computer and play with the colors? IMO we should use `$fish_colors_*` in order to make sure the theme is going to blend with the shell colors.

**Comment by bpinto** [comment]
@nesl247 this is why I think we should use the `$fish_colors`:
![screen shot 2015-08-28 at 19 16 44](https://cloud.githubusercontent.com/assets/526122/9553804/62267fb4-4db9-11e5-9d6e-3a175e786d6a.png)

The colors are ugly IMO, but I know it depends on your colorscheme.

**Comment by bucaran** [comment]
LGTM

Using $fish_colors :+1:

**Comment by jonscottclark** [comment]
@bpinto Even though we can assume that most users of oh-my-fish will be "advaned" users in some sense, and will get what's happening, I just think that a default theme for this framework shouldn't deviate too much from the expected behaviour of a regular bash prompt.

It seems like I am just choosing this one issue to comment on, but it's only the only time that I have been able to read some of these issues! You and @bucaran are just too fast :smile:

**Comment by bobthecow** [comment]
i really like abbreviated paths, and i feel like that's one of the things that stood out to me most after switching from bash.

**Comment by bpinto** [comment]
@jonscottclark You have been with us for years now! :P Glad to hear another feedback from you.

@bobthecow Should we aim for `~/.l/s/omf` style instead?

**Comment by bucaran** [comment]
`~/.l/s/omf` please.

**Comment by bobthecow** [comment]
if we're going to make the prompt git-aware, we should use the bobthefish style abbreviated path.

`~/a/b/[ project +- ]/c/d/folder`

**Comment by bobthecow** [comment]
![screenshot 2015-08-28 11 30 10 2](https://cloud.githubusercontent.com/assets/53660/9554080/336f3878-4d78-11e5-9c81-c9ca806f8be2.png)

**Comment by bpinto** [comment]
@bobthecow I can't imagine how it would work. Perhaps this could be a PR to improve the theme?

**Comment by bobthecow** [comment]
just steal the code from bobthefish :)

**Comment by bobthecow** [comment]
https://github.com/oh-my-fish/theme-bobthefish/blob/master/fish_prompt.fish#L99-L122

**Comment by bpinto** [comment]
@bobthecow I couldn't steal it, will have to stop and understand the code. :(

**Comment by bobthecow** [comment]
those 3 (4) functions just return the segments you need.

`~/stuff/before/project/name` (but abbreviated, obvs).

`projectname`

`stuff/after/project/name`

… there are two of the middle functions, one for hg, one for git.

**Comment by bpinto** [comment]
Thanks, I'll have a look at it again tomorrow.

**Comment by derekstavis** [comment]
I think this is closed as #542 is merged?

**Comment by bpinto** [comment]
Its kind of. We still need to make it the default and add this change bobthecow suggested. But it's certainly a new issue.
